docs/pages/api/automatically-register-agents.mdx

```diff
@@ -0,0 +1,988 @@
+---
+title: Automatically Register Resources with Teleport
+description: Learn how to use the Teleport API to start agents automatically when you add resources to your infrastructure.
+---
+
+You can use Teleport's API to automatically register resources in your
+infrastructure with your Teleport cluster. 
+
+Teleport already supports the automatic discovery of [Kubernetes
+clusters](../kubernetes-access/discovery.mdx) in AWS, Azure, and Google Cloud,
+as well as [servers](../server-access/guides/ec2-discovery.mdx) on Amazon EC2.
+To support other resources and cloud providers, you can use the API to write
+your own workflow.
+
+In this guide, we will demonstrate some libraries you can use to automatically
+register resources with Teleport. We will use an example you can run locally on
+your workstation.
+
+Automatic registration consists of the following steps:
+
+- Look up resources in your infrastructure from a service discovery solution,
+  e.g., the Kubernetes API server, Consul, or your cloud provider's APIs.
+- Use the Teleport gRPC API to look up resources registered with Teleport.
+- For any resources in your infrastructure that are not registered with
+  Teleport, use the Teleport API to spin up a new Teleport service or register
+  the resource with an exisitng Teleport service.
+- For any resources that are registered with Teleport but not in your
+  infrastructure, use the Teleport API to deregister the resource and, if
+  necessary, remove the Teleport service proxying the resource.
```

Comment:
In the desktop registration [example](https://github.com/gravitational/teleport/tree/master/examples/desktop-registration) I've instead allowed items to expire automatically, so this step isn't necessary. I find it makes the example easier, but up to you.

Reply:
I wanted to make the removal close to immediate for the demonstration of the app in the final section, but I'll add a note that you can change the TTL.